nvflare/lighter/poc.py

```diff
@@ -34,38 +32,58 @@ def clone_client(num_clients: int):
     shutil.rmtree(src_folder)
 
 
-def main():
-    parser = argparse.ArgumentParser()
-    parser.add_argument("-n", "--num_clients", type=int, default=1, help="number of client folders to create")
+def clone_poc_folder(src_poc_folder, dest_poc_folder):
+    shutil.rmtree(dest_poc_folder, ignore_errors=True)
+    try:
+        shutil.copytree(src_poc_folder, dest_poc_folder)
+    except BaseException as e:
+        print(f"Unable to copy poc folder from {src_poc_folder}, Exit. {e} ")
+        exit(1)
+    for root, dirs, files in os.walk(dest_poc_folder):
+        for file in files:
+            if file.endswith(".sh"):
+                os.chmod(os.path.join(root, file), 0o755)
 
-    args = parser.parse_args()
 
+def get_src_poc_parent():
     file_dir_path = pathlib.Path(__file__).parent.absolute()
-    poc_zip_path = file_dir_path.parent / "poc.zip"
-    poc_folder_path = file_dir_path.parent / "poc"
+    return file_dir_path.parent
+
+
+def get_src_poc_dir():
+    return os.path.join(get_src_poc_parent(), "poc")
+
+
+def get_dest_poc_dir(dest_poc):
+    if not dest_poc:
+        dest_poc = os.path.join(os.getcwd(), "poc")
+    return dest_poc
+
+
+def prepare_poc(num_clients, poc_workspace):
     answer = input("This will delete poc folder in current directory and create a new one. Is it OK to proceed? (y/N) ")
     if answer.strip().upper() == "Y":
-        dest_poc_folder = os.path.join(os.getcwd(), "poc")
-        shutil.rmtree(dest_poc_folder, ignore_errors=True)
-        try:
-            shutil.unpack_archive(poc_zip_path)
-        except shutil.ReadError:
-            print(f"poc.zip not found at {poc_zip_path}, try to use template poc folder")
-            try:
-                shutil.copytree(poc_folder_path, dest_poc_folder)
-            except BaseException:
-                print(f"Unable to copy poc folder from {poc_folder_path}.  Exit")
-                exit(1)
-        for root, dirs, files in os.walk(dest_poc_folder):
-            for file in files:
-                if file.endswith(".sh"):
-                    os.chmod(os.path.join(root, file), 0o755)
-        clone_client(args.num_clients)
-        print("Successfully creating poc folder.  Please read poc/Readme.rst for user guide.")
-        print("Note POC does not have features of overseer and server switch over.")
+        dest_poc_folder = get_dest_poc_dir(poc_workspace)
+        src_poc_folder = get_src_poc_dir()
+        clone_poc_folder(src_poc_folder=src_poc_folder, dest_poc_folder=dest_poc_folder)
+        clone_client(num_clients, dest_poc_folder)
+        print(f"Successfully creating poc folder at {dest_poc_folder}.  Please read poc/Readme.rst for user guide.")
```

Comment:
@IsaacYangSLA @chesterxgchen does this mean we don't need/pack "poc.zip" anymore?